torch/_subclasses/meta_utils.py

```diff
@@ -1780,12 +1841,11 @@ def is_c_of_r(
             if t.is_parameter:
                 r._is_param = True
 
+            # TODO(soulitzer): update this note
             # See Note: [Creating symbolic nested int]
             if t.nested_int is not None:
                 assert _is_fake_tensor(r)
-                r.nested_int_memo = r.fake_mode.create_symbolic_nested_int(
-                    nt_tensor_id=t.nested_int
-                )
+                r.register_nested_int_id(t.nested_int)
```

Comment:
hmm, two things I'm trying to understand:

(1) it looks like we store this `nested_int_id` directly on `FakeTensor` - I guess this is historic, and I can't remember why, but logically it seems to me like `nested_int_id` should live on either `NJT` or `CachedTensor`, instead of directly on a `FakeTensor`. Do you remember why that's the case?

(2) The `register_nested_int_id` API as written accepts an id as an optional argument, and generates one (from some nonlocal state) otherwise. But looking at all the call-sites, it looks like you always pass in an id. Do we need the no-argument overload?